src/ansys/systemcoupling/core/adaptor/impl/injected_commands.py

```diff
@@ -100,23 +130,87 @@ def _wrap_add_participant(
+    case._clear_state(**kwargs)
 
 
-def _wrap_solve(root_object: Container, part_mgr: ParticipantManager) -> None:
+def _wrap_solve(solution: Container, part_mgr: ParticipantManager) -> None:
     if part_mgr is None:
-        root_object._solve()
+        solution._solve()
     else:
         part_mgr.solve()
 
 
+def _ensure_file_available(session: SessionProtocol, filepath: str) -> str:
+    """If we are in a "PIM" session, copies the file specified by ``filepath``
+    into the working directory, so that it is available for download.
+
+    A suffix is added to the name of the copy to make the name unique, and
+    the new name is returned.
+    """
+    # Note: it is a general issue with files in a PIM session that they can
+    # only be uploaded to/downloaded from the root working directory. We
+    # might want to consider integrating something like this directly into
+    # the file_transfer module later so that it is more seamless.
+
+    if not pypim.is_configured():
+        return filepath
+
+    # Copy file to a unique name in the working directory
+    file_name = os.path.basename(filepath)
+    root_name, _, ext = file_name.rpartition(".")
+    ext = f".{ext}" if ext else ""
+    new_name = f"{root_name}_{int(time.time())}_{random.randint(1, 10000000)}{ext}"
+
+    session._native_api.ExecPythonString(
+        PythonString=f"import shutil\nshutil.copy('{filepath}', '{new_name}')"
+    )
+
+    session.download_file(new_name, ".")
+    return new_name
+
+
+def show_plot(session: SessionProtocol, rpc, **kwargs):
+    setup = session.setup
+    working_dir = kwargs.pop("working_dir", ".")
+    interface_name = kwargs.pop("interface_name", None)
+    if interface_name is None:
+        interfaces = setup.coupling_interface.get_object_names()
+        if len(interfaces) == 0:
+            return
+        if len(interfaces) > 1:
+            raise RuntimeError(
+                "show_plot() currently only supports a single interface."
+            )
+        interface_name = interfaces[0]
+    interface_object = setup.coupling_interface[interface_name]
+    interface_disp_name = interface_object.display_name
+
+    transfer_names = kwargs.pop("transfer_names", [])
+    if not transfer_names:
+        transfer_names = interface_object.data_transfer.get_object_names()
+    transfer_disp_names = [
+        interface_object.data_transfer[trans_name].display_name
+        for trans_name in transfer_names
+    ]
+    # TODO : better way to do this?
+    is_transient = setup.solution_control.time_step_size is not None
+
+    file_path = _ensure_file_available(
+        session, os.path.join(working_dir, "SyC", f"{interface_name}.csv")
+    )
+
+    return create_and_show_plot(
+        is_transient,
+        [(interface_name, interface_disp_name, transfer_disp_names)],
+        [file_path],
+    )
+
+
 def get_injected_cmd_data() -> list:
     """Gets a list of injected command data in the right form to insert
```

Comment:
```suggestion
    """Get a list of injected command data in the right form to insert
```
Descriptions for a function or method should not have an "s" or "es" at the end of the initial verb in the description.